### PR TITLE
Logics of spanning configuration of TokenHolderRevenueFundContractEnsemble

### PR DESCRIPTION
**Pull request by jijordre**
### Description

This PR corrects a few items in the spanning of configurations of `TokenHolderRevenueFundContract` instances. Also it adjusts the claimable and withdrawable fees output from functions of `FeesClaimant`.

### Issues

Issues: #157

### Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] The [Contribution Guidelines][1] has been followed
- [x] Tests for the changes have been added and all test pass (`npm test`)
- [x] JSDoc comments have been reviewed and added / updated as needed
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
- [x] Package version number was updated according to semantic rules
- [x] Any changes to upstream dependencies have already been published

[1]: https://github.com/hubiinetwork/nahmii-sdk#contributing

### CODE REVIEW

File: Docs/fees-claimant.md [hunk 1]
```diff
@@ -7,17 +7,17 @@ A class for the claiming and withdrawal of accrued fees
 **Kind**: global class  
 
 * [FeesClaimant](#module_nahmii-sdk)
-    * [.claimableAccruals(wallet, currency, [startAccrual], [endAccrual])](#module_nahmii-sdk+claimableAccruals) ⇒ <code>Promise</code>
-    * [.claimableFeesForAccruals(wallet, currency, startAccrual, endAccrual)](#module_nahmii-sdk+claimableFeesForAccruals) ⇒ <code>Promise</code>
-    * [.claimFeesForAccruals(wallet, currency, startAccrual, endAccrual, [options])](#module_nahmii-sdk+claimFeesForAccruals) ⇒ <code>Promise</code>
-    * [.claimableFeesForBlocks(wallet, currency, startBlock, endBlock)](#module_nahmii-sdk+claimableFeesForBlocks) ⇒ <code>Promise</code>
-    * [.claimFeesForBlocks(wallet, currency, startBlock, endBlock, [options])](#module_nahmii-sdk+claimFeesForBlocks) ⇒ <code>Promise</code>
+    * [.claimableAccruals(wallet, currency, [firstAccrual], [lastAccrual])](#module_nahmii-sdk+claimableAccruals) ⇒ <code>Promise</code>
```
Comment (hubiierik) [inline Docs/fees-claimant.md:10]: First/Last: Love it!

File: lib/claim/fees-claimant.js [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 const TokenHolderRevenueFundContractsEnsemble = require('./token-holder-revenue-fund-contracts-ensemble');
 const NestedError = require('../nested-error');
+const {ethers: {utils: {formatUnits}}} = require('ethers');
```
Comment (hubiierik) [inline lib/claim/fees-claimant.js:5]: Pedantic comment: Can also be written more tersely
`const {formatUnits} = require('ethers').utils;`

File: lib/claim/fees-claimant.js [hunk 3]
```diff
@@ -55,13 +58,19 @@ class FeesClaimant {
...
-        return ensemble.claimableAmountByAccruals(
+        const provider = _provider.get(this);
+
+        const claimableAmount = await ensemble.claimableAmountByAccruals(
             wallet, currency, firstAccrual, lastAccrual
         );
+
+        const tokenInfo = await provider.getTokenInfo(currency.ct.toString(), true);
+
+        return tokenInfo ? formatUnits(claimableAmount, tokenInfo.decimals) : claimableAmount.toString();
```
Comment (hubiierik) [inline lib/claim/fees-claimant.js:73]: At what situations do we expect `tokenInfo` to be undefined?
If `tokenInfo` is undefined, are we sure that it makes sense to just convert  `claimableAmount` to a string?
If not, would i make sense to throw an Error?

File: lib/claim/fees-claimant.js [hunk 4]
```diff
@@ -127,11 +142,17 @@ class FeesClaimant {
...
+     * @returns {Promise} A promise that resolves into a string value representing the withdrawable amount of fees.
      */
     async withdrawableFees(wallet, currency) {
         const ensemble = _tokenHolderRevenueFundContractEnsemble.get(this);
-        return ensemble.stagedBalance(wallet, currency);
+        const provider = _provider.get(this);
+
+        const withdrawableFees = await ensemble.stagedBalance(wallet, currency);
+
+        const tokenInfo = await provider.getTokenInfo(currency.ct.toString(), true);
```
Comment (hubiierik) [inline lib/claim/fees-claimant.js:153]: These two lines of currency-formatting has been repeated three times in the code above.
Would that make it a candidate for factoring out into a utility function?

File: lib/claim/token-holder-revenue-fund-contracts-ensemble.js [hunk 5]
```diff
@@ -18,33 +18,41 @@ const minBigNumber = (n1, n2) => n1.lt(n2) ? n1 : n2;
...
-            (await configuration.contract.closedAccrualsByCurrency(
-                currency.ct.toString(), currency.id, span.lastAccrual
-            )).endBlock.toNumber();
+        if (0 === closedAccrualsCount)
+            continue;
+
+        const firstBlock = (await configuration.contract.closedAccrualsByCurrency(
+            currency.ct.toString(), currency.id, 0
+        )).startBlock.toNumber();
+        const lastBlock = (await configuration.contract.closedAccrualsByCurrency(
```
Comment (hubiierik) [inline lib/claim/token-holder-revenue-fund-contracts-ensemble.js:37]: Pedantic comment: Would it make sense to insert a blank line before `const lastBlock = ...`?

File: Docs/fees-claimant.md (hunk 1 above)
Comment (jijordre) [reply Docs/fees-claimant.md:10]: 😆

File: lib/claim/fees-claimant.js (hunk 2 above)
Comment (jijordre) [reply lib/claim/fees-claimant.js:5]: I reckon that in that case you can just as well go all the way to
```
const formatUnits = require('ethers').utils.formatUnits;
```

I reckon my preference for the initial one is the use of destructuring multiple variables on the same call to `require()`.

File: lib/claim/token-holder-revenue-fund-contracts-ensemble.js [hunk 6]
```diff
@@ -269,13 +286,16 @@ class TokenHolderRevenueFundContractsEnsemble {
...
+        const configurations = _configurations.get(this);
         const currencyKey = currencyToKey(currency);
 
         try {
             const txs = [];
 
-            for (const configuration of ensemble) {
+            for (const configuration of configurations) {
+                if (!configuration.spansByCurrency || !configuration.spansByCurrency.has(currencyKey))
+                    continue;
```
Comment (hubiierik) [inline lib/claim/token-holder-revenue-fund-contracts-ensemble.js:297]: Is it not the the case that `await span.call(...)` above ensures that span exists?
If the span is ensured to exist, do we need this test here?
Same comment goes for the other two copies of the same check.

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nahmii-sdk",
-  "version": "4.5.0",
+  "version": "4.5.1",
```
Comment (hubiierik) [inline package.json:3]: I can see the argument for only changing the patch number despite the fact that the PR has introduced API changes.
So maybe that is the right thing to do.

File: lib/claim/fees-claimant.js (hunk 3 above)
Comment (jijordre) [reply lib/claim/fees-claimant.js:73]: There is a parallel case in https://github.com/hubiinetwork/nahmii-sdk/blob/master/lib/wallet/wallet.js#L82. Albeit in that case there is no conversion of BN amount to string. At least that case suggests a scenario in which there are monetary amounts (through balances) in nahmii that are not of supported token types.

File: lib/claim/token-holder-revenue-fund-contracts-ensemble.js (hunk 5 above)
Comment (jijordre) [reply lib/claim/token-holder-revenue-fund-contracts-ensemble.js:37]: 👍

File: lib/claim/fees-claimant.js (hunk 4 above)
Comment (jijordre) [reply lib/claim/fees-claimant.js:153]: 👍

File: package.json (hunk 7 above)
Comment (jijordre) [reply package.json:3]: That is my rationale too.

File: lib/claim/token-holder-revenue-fund-contracts-ensemble.js (hunk 6 above)
Comment (jijordre) [reply lib/claim/token-holder-revenue-fund-contracts-ensemble.js:297]: True

**Pull request merged**